packages/js/src/utils/npm-config.ts

```diff
@@ -0,0 +1,109 @@
+import { joinPathFragments } from '@nx/devkit';
+import { exec } from 'child_process';
+import { existsSync } from 'fs';
+import { PackageJson } from 'nx/src/utils/package-json';
+
+export async function parseRegistryOptions(
+  cwd: string,
+  pkg: {
+    root: string;
+    manifest: PackageJson;
+    manifestPath: string;
+  },
+  options: {
+    registry?: string;
+    tag?: string;
+  },
+  logWarnFn: (message: string) => void = console.warn
+): Promise<{ registry: string; tag: string; registryConfigKey: string }> {
+  if (existsSync(joinPathFragments(pkg.root, '.npmrc'))) {
+    logWarnFn(
+      `Ignoring .npmrc file detected in the package root. Nested .npmrc files are not supported by npm. Only the .npmrc file at the root of the workspace will be used. To customize the registry or tag for specific packages, see https://nx.dev/recipes/nx-release/configure-custom-registries.`
+    );
+  }
+
+  const scope = pkg.manifest.name.startsWith('@')
+    ? pkg.manifest.name.split('/')[0]
+    : '';
+
+  // If the package is scoped, then the registry argument that will
+  // correctly override the registry in the .npmrc file must be scoped.
+  const registryConfigKey = scope ? `${scope}:registry` : 'registry';
+
+  const publishConfigRegistry = pkg.manifest.publishConfig?.[registryConfigKey];
+
+  // Even though it won't override the actual registry that's actually used,
+  // the user might think otherwise, so we should still warn if the user has
+  // set a 'registry' in 'publishConfig' for a scoped package.
+  if (publishConfigRegistry || pkg.manifest.publishConfig?.registry) {
+    logWarnFn(
+      `Registry detected in the 'publishConfig' ${pkg.manifestPath}. Configuring the registry this way prevents it from being overridden via the --registry argument. To customize the registry or tag for specific packages in a way that can be overridden, see https://nx.dev/recipes/nx-release/configure-custom-registries.`
+    );
+  }
+
+  const registry =
+    // `npm publish` will always use the publishConfig registry if it exists, even over the --registry arg
```

Comment:
Since we can't override this behavior, we just warn the user and go along with it.